lib/Dialect/Comb/CombFolds.cpp

```diff
@@ -893,6 +893,26 @@ LogicalResult MuxOp::canonicalize(MuxOp op, PatternRewriter &rewriter) {
       return success();
     }
   }
+
+  // mux(selector, x, mux(selector, y, z) = mux(selector, x, z)
+  if (auto falseCase = dyn_cast_or_null<MuxOp>(op.falseValue().getDefiningOp())) {
+    if (op.cond() == falseCase.cond()) {
+      const auto newT = op.trueValue();
```

Comment:
Please drop the "const"s for local variables.  Also, I'd recommend spelling out "Value" instead of auto.  It is just one more character and increases clarity a bit.

Reply:
Done!